### DISCUSSION TITLE
How to use the libreSBC

### DISCUSSION THREAD

**Discussion by bcahyadi** [hnimminh/libresbc#39] [Q&A] [answered]
Hi,

I have done with the installation. My question is how to config the tool? E.g I have a freepbx,a service provider and libreSBC. How to config the sip trunk, call routing for inbound and outbound?
Service provider<--->LibreSBC<--->FreePBX

thanks

**Comment by hnimminh** [comment, marked as answer]
Hi @bcahyadi , 
After installation, you can find the API doc via this link https://your-libresbc-ip:8443/apidocs
In generally, to make libresbc work you will need to define:
* cluster node
* netalias
* media
* capacity
For the flow `Service Provider ---> LibreSBC --->FreePBX` You will need:
* gateway = `freepbx`
* intcon:out for outbound traffic to `freepbx`
* routing
* incon:in to accept traffic from `service provider`

**Reply by bcahyadi** [reply]
hi @hnimminh  thanks for the reply,

where to define all the things from the LibreSBC server? which directory and which file needs to configure. is it need an external tool to config since no GUI for the LibreSBC?

**Reply by hnimminh** [reply]
Since GUI is not developed, You will need use the `curl` or `postman` ( or any kind of http client) to config via HTTP API.

**Reply by bcahyadi** [reply]
hi @hnimminh 

need your advice I could not update the config with api
client app: postman

![image](https://user-images.githubusercontent.com/9133392/145329943-bc0ad17c-24e3-4bd8-a371-ec181cad12b0.png)

**Reply by hnimminh** [reply]
@bcahyadi 
The member must is a item in list of candidates, to get list of candidates please call the **PREDEFINE** API /libreapi/predefine, 
example:

```
curl -X GET 'https://127.0.0.1:8443/libreapi/predefine'
  {
    "application": "LIBRESBC",
    "swversion": "0.5.7",
    "description": "Open Source Session Border Controller for Large-Scale Voice Infrastructures",
    "nodeid": "apollo",
    "candidates": [
        "apollo"
    ],
    "codecs": [
        "ALAW",
        "ULAW",
        "OPUS",
        "G729",
        "AMR",
        "AMR-WB"
    ]
}
```

you see the `apollo` in `candidates`, so `apollo` should be input into your request.

Hope this help!
